File: src/main/java/uk/edu/populationfitness/models/genes/Ridge/Schwefel220Genes.java
package uk.edu.populationfitness.models.genes.Ridge;

import uk.edu.populationfitness.models.Config;
import uk.edu.populationfitness.models.genes.bitset.NormalizingBitSetGenes;

public class Schwefel220Genes extends NormalizingBitSetGenes {
    public Schwefel220Genes(Config config) {
        super(config, 10.0);
    }

    @Override
    protected double calculateNormalizationRatio(int n) {
        return 10.0 * n;
    }

    @Override
    protected double calculateFitnessFromGenes(double[] unknowns) {

        /*
          f left (x right ) =- sum from {i=1} to {n} {left lline {x} rsub {i} right rline}
         */
        double fitness = 10.0 * unknowns.length;

        for (double x : unknowns) {
            fitness -= Math.abs(x);
        }

        return fitness;
    }
}

File: DotNet/PopulationFitness/PopulationFitness/Models/Genes/Ridge/Schwefel220Genes.cs
using PopulationFitness.Models.Genes.BitSet;
using System;

namespace PopulationFitness.Models.Genes.Ridge
{
    public class Schwefel220Genes : NormalizingBitSetGenes
    {
        public Schwefel220Genes(Config config) : base(config, 10.0)
        {
        }

        protected override double CalculateNormalizationRatio(int n)
        {
            return 10.0 * n;
        }

        protected override double CalculateFitnessFromIntegers(long[] integer_values)
        {

            /*
              f left (x right ) =- sum from {i=1} to {n} {left lline {x} rsub {i} right rline}
             */
            double fitness = 10.0 * integer_values.Length;

            foreach (long integer_value in integer_values)
            {
                double x = Interpolate(integer_value);
                fitness -= Math.Abs(x);
            }

            return fitness;
        }
    }
}
